### PR TITLE
Refactor class RenderJob in general

### PR DESCRIPTION
**Pull request by sschulz92**
- to avoid NullPointerExceptions
- to be less complex by extracting functionality into functions

### CODE REVIEW

File: plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RenderJob.java [hunk 1]
```diff
@@ -197,14 +231,11 @@ protected void postRendering() {
...
-    /**
-     * @see org.eclipse.core.runtime.jobs.Job#run(org.eclipse.core.runtime.IProgressMonitor)
-     */
     @Override
     protected IStatus run(IProgressMonitor monitor) {
 
-        if (!((RenderManager) getExecutor().getContext().getRenderManager()).isRenderingEnabled()) {
-            return Status.OK_STATUS;
+        if (!isRenderingEnabled()) {
+            return Status.CANCEL_STATUS;
```
Comment (fgdrf) [inline plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RenderJob.java:238]: It the layer marked somehow as "not rendered". Is the canceling status an indicator for an "error". Can you describe implications of this switch?

File: plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RenderJob.java [hunk 2]
```diff
@@ -282,6 +314,10 @@ public void setBounds(ReferencedEnvelope bounds) {
         this.bounds = bounds;
     }
 
+    public void disposeRenderExecutor() {
```
Comment (fgdrf) [inline plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RenderJob.java:317]: its not API  and its not called internally. What is this method for?

File: plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RenderJob.java [hunk 3]
```diff
@@ -328,12 +364,20 @@ public boolean isConflicting(ISchedulingRule rule) {
...
+                    && !other.isCanceled()) {
+
+                if (isUsingLocalResources(getRenderer())
+                        && isUsingLocalResources(other.getRenderer())) {
+                    return true;
+                }
+                if (isConflictingGeoResources(other.getRenderer())) {
+                    return true;
+                }
+            }
```
Comment (fgdrf) [inline plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RenderJob.java:378]: Please correct me if I'm wrong. You added another check `isConflictingGeoResources`  ..

File: plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RenderJob.java [hunk 4]
```diff
@@ -345,6 +389,20 @@ private boolean isUsingLocalResources(Renderer renderer) {
...
+                        .getPersistentProperties().get("RENDERING_CONFLICT_RESOLUTION_MARKER"); //$NON-NLS-1$
+                String conflictResolutionMarker2 = (String) getRenderer().getContext()
+                        .getGeoResource().getPersistentProperties()
+                        .get("RENDERING_CONFLICT_RESOLUTION_MARKER"); //$NON-NLS-1$
+                return (conflictResolutionMarker1 != null && conflictResolutionMarker2 != null
+                        && conflictResolutionMarker1.equals(conflictResolutionMarker2));
+            } catch (Exception e) {
+                return false;
+            }
+        }
```
Comment (fgdrf) [inline plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RenderJob.java:404]: .. which checks for 'RENDERING_CONFLICT_RESOLUTION_MARKER' value. Where is the key used to set it in perstistentProperties of GeoResource? I guess I miss something ..

**Review by fgdrf — COMMENTED** [review]
What is the intention of this refactoring? I guess some are for better readability, some might address optimizations in case of layer should not be rendered for current scale...

File: plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RenderJob.java (hunk 1 above)
Comment (sschulz92) [reply plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RenderJob.java:238]: I revert this change in line 238.

File: plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RenderJob.java (hunk 2 above)
Comment (sschulz92) [reply plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RenderJob.java:317]: I will revert this.

File: plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RenderJob.java (hunk 3 above)
Comment (sschulz92) [reply plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/render/impl/RenderJob.java:378]: I will revert the addition of `isConflictingGeoResources()`

**Pull request merged**